scrape/scrape.go

```diff
@@ -47,6 +47,11 @@ import (
 	"github.com/prometheus/prometheus/storage"
 )
 
+// Temporary tolerance for scrape appends timestamps alignment, to enable better
+// compression at the TSDB level.
+// See https://github.com/prometheus/prometheus/issues/7846
+const scrapeTimestampTolerance = 2 * time.Millisecond
```

Comment:
Perhaps make this a fraction of the scrape interval? Like one percent or even just one permil of the scrape interval.

Reply:
1% of 60s is 600ms, which is rather a lot of jitter and well beyond that a user could reasonably be expected to be robust to. I think a small flat number like this is fine.

Reply:
Yeah we try to protect from small useless jitters, more that 2ms might be a too big 'lie'. Think about e.g. the system time exposed by the node_exporter : `node_time_seconds-timestamp(node_time_seconds)`.

Reply:
I had cases of sub-second scrape intervals, where 2ms might be too much. The Go runtime can actually act in the sub-ms domain at smaller scales.

Reply:
The bug that prompted this means that Go can no longer operate precisely in those sorts of time ranges, plus there's a cutoff if the scrape interval is under 200ms so this will never be more than 1%.

2ms is also a fair bit below the kernel scheduling quantum, and a TCP retransmit takes far longer to handle.

Reply:
If the parameters make sure this will never adjust by more than 1% of the scrape interval, I guess it's fine.

(I'd still prefer to allow switching off the feature.)